packages/components/src/flex/flex/hook.js

```diff
@@ -78,34 +78,43 @@ export function useFlex( props ) {
 			marginBottom: wrap ? `calc(${ space( gap ) } * -1)` : undefined,
 		} );
 
-		sx.Items = css( {
-			/**
-			 * Workaround to optimize DOM rendering.
-			 * We'll enhance alignment with naive parent flex assumptions.
-			 *
-			 * Trade-off:
-			 * Far less DOM less. However, UI rendering is not as reliable.
-			 */
-			'> * + *:not(marquee)': {
-				marginTop: isColumn ? space( gap ) : undefined,
-				marginRight: ! isColumn && isReverse ? space( gap ) : undefined,
-				marginLeft:
-					! isColumn && ! isReverse ? space( gap ) : undefined,
-			},
-		} );
+		/**
+		 * Workaround to optimize DOM rendering.
+		 * We'll enhance alignment with naive parent flex assumptions.
+		 *
+		 * Trade-off:
+		 * Far less DOM less. However, UI rendering is not as reliable.
+		 */
+		sx.Items = css`
+			> * + *:not( marquee ) {
+				margin-top: ${ isColumn ? space( gap ) : undefined };
+				${ rtl( {
+					marginLeft:
+						! isColumn && ! isReverse ? space( gap ) : undefined,
+					marginRight:
+						! isColumn && isReverse ? space( gap ) : undefined,
+				} )() }
```

Comment:
After discussing with @sarayourfriend , we agreed that it is best not to immediately execute the function returned by `rtl()`.

Emotion, in fact, will [recognize the function and interpolate the returned value every time the props update](https://emotion.sh/docs/styled#changing-based-on-props) (the `props` will be also passed as an argument to the interpolated function, although that doesn't matter in this case).

This means that the styles wrapped in the `rtl` function will be able to be re-computed in case the page's language gets updated at runtime.

```suggestion
				${ rtl( {
					marginLeft:
						! isColumn && ! isReverse ? space( gap ) : undefined,
					marginRight:
						! isColumn && isReverse ? space( gap ) : undefined,
				} ) }
```

Reply:
Thanks! Didn't know that.

Reply:
Seem like typescript wasn't to happy with the change 🤔

Reply:
Revertet these changes. Correct me if I'm wrong, but it looks like it only applies to `styled` and not to `css` as used here?

Reply:
You are indeed correct, that was my bad for not noticing this was a `css` function call and not a `styled` component.

We've just opened and merged #33882 which adds a couple of improvements to the `rtl()` function. You should be able to rebase on top of `trunk` and add `rtl.watch()` to the list of dependencies of the `useMemo` hook (currently that would be on line 135 of this file).

This would ensure that rtl styles get re-calculated when there's a change of RTL/LTR locale. Sorry again for the wrong suggestion!

Reply:
No worries. An enhancement was added because of it 😄 

I'll rebase and implement the changes needed.